swap/src/monero/wallet_rpc.rs

```
@@ -146,12 +148,18 @@ impl WalletRpc {
 
         let mut reader = BufReader::new(stdout).lines();
 
-        while let Some(line) = reader.next_line().await? {
-            if line.contains("Starting wallet RPC server") {
-                break;
-            }
+        // If we do not hear from the monero_wallet_rpc process for 3 seconds we assume
+        // it is is ready
+        while let Ok(res) = tokio::time::timeout(Duration::from_secs(3), reader.next_line()).await {
+            res?;
         }
 
+        // Send a json rpc request to make sure monero_wallet_rpc is ready
+        Client::localhost(port)
+            .get_version()
+            .await
+            .expect("monero wallet rpc is ready to accept json rpc requests");
```

Why are we panicking here instead of returning a Result?